worker/process-image.js

```diff
@@ -36,20 +37,44 @@ function _processImage (s3, scene, url, key, cb) {
       url = `https://www.googleapis.com/drive/v3/files/${pieces[1]}?alt=media&key=${config.gdriveKey}`;
     }
 
-    log(['debug'], 'Downloading ' + url + ' to ' + path);
-
     var downloadStatus;
-    request(url)
-    .on('response', function (resp) { downloadStatus = resp.statusCode; })
-    .on('error', callback)
+    var upload;
+    var stream;
+    var messages = [];
+    // Open local read stream if file was uploaded
+    if (url.includes("file://")) {
+      upload = './uploads/' + urlTools.parse(url).host;
+      log(['debug'], 'Transferring ' + upload + ' to ' + path);
+      stream = fs.createReadStream(upload);
+    } else {
+      // Open a download stream if file is remote
+      log(['debug'], 'Downloading ' + url + ' to ' + path);
+      stream = request(url);
+      stream.on('response', function (resp) { downloadStatus = resp.statusCode; });
+      stream.on('error', callback);
+    }
+
+    // Write the stream
+    stream
     .pipe(fs.createWriteStream(path))
     .on('finish', function () {
       if (downloadStatus < 200 || downloadStatus >= 400) {
         return callback(new Error('Could not download ' + url +
          '; server responded with status code ' + downloadStatus));
       }
+      // Cleanup local files
+      if (url.includes("file://")) {
+        fs.stat(upload, function (err, stats) {
```

Comment:
There's no need to check for the file existence.
Just try to remove it and handle the error if it fails.

It's also suggested by the docs: https://nodejs.org/api/fs.html#fs_fs_stat_path_callback